freqtrade/freqai/RL/BaseEnvironment.py

```diff
@@ -188,7 +195,7 @@ def _get_observation(self):
         """
         features_window = self.signal_features[(
             self._current_tick - self.window_size):self._current_tick]
-        if self.add_state_info:
+        if self.add_state_info and self.live:
```

Comment:
isn't this (additional) check redundant, considering the checks in line 85 - which deactivate `add_state_info` ?

Reply:
Yes, in the current for it is. But I wrote this for explicit readability to other developers. i.e. "we CANNOT use state info unless we are live." Further, if a user decides to take control of `self.add_state_info` somewhere in their own custom model, then this check will no longer be redundant.